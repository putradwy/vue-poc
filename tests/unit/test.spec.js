import { mount } from '@vue/test-utils'
import MoviePreview from '@/components/Moviepreview.vue'
import MovieList from '@/components/Movielist.vue'

describe('Moviepreview.vue', () => {
  it('components should render', () => {
    const preview = {
        image: "Movie image",
        title: "Movie title",
        description: "Movie Descrtiption"
    }
    const wrapper = mount(MoviePreview, {
      props: preview
    })
    expect(wrapper.html()).toMatchSnapshot();
  })
})

describe('Movielist.vue', () => {
  it('components should render', () => {
    const lists = [
      {
        "title": "movie 1",
        "id": "1"
      },
      {
        "title": "movie 2",
        "id": "2"
      }
    ]
    const wrapper = mount(MovieList, {
      data() {
        return {
          lists,
          value: {
            id: "1"
          }
        }
      }
    })
    expect(wrapper.html()).toMatchSnapshot();
  })

  it("should movieList handle onClick", async () => {
    const lists = [
      {
        "title": "movie 1",
        "id": "1"
      },
      {
        "title": "movie 2",
        "id": "2"
      }
    ]
    const wrapper = mount(MovieList, {
      data() {
        return {
          lists,
          value: {
            id: "1"
          }
        }
      },
    })
    await wrapper.find("[id='item-2']").trigger("click");
    expect(wrapper.html()).toMatchSnapshot();
  });
})
