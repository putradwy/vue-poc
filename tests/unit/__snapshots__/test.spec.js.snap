// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movielist.vue components should render 1`] = `
<div class="movie-list">
  <div class="list-container">
    <ul>
      <ol class="active" id="item-1">movie 1</ol>
      <ol class="" id="item-2">movie 2</ol>
    </ul>
  </div>
</div>
`;

exports[`Movielist.vue should movieList handle onClick 1`] = `
<div class="movie-list">
  <div class="list-container">
    <ul>
      <ol class="" id="item-1">movie 1</ol>
      <ol class="active" id="item-2">movie 2</ol>
    </ul>
  </div>
</div>
`;

exports[`Moviepreview.vue components should render 1`] = `
<div class="movie-preview-container">
  <div class="preview-image"><img src="Movie image"></div>
  <h3 class="preview-title">Movie title</h3>
  <div class="preview-description">Movie Descrtiption</div>
</div>
`;
